src/createReconciler.lua

```diff
@@ -100,6 +122,18 @@ local function createReconciler(renderer)
 					virtualNode.legacyContext
 				)
 
+				-- If updating this node has caused a component higher up the tree to re-render
+				-- and updateChildren to be re-entered for this virtualNode then
+				-- this result is invalid and needs to be disgarded.
+				if config.tempFixUpdateChildrenReEntrancy then
+					if virtualNode.updateChildrenCount ~= currentUpdateChildrenCount then
+						if childNode then
+							unmountVirtualNode(childNode)
+						end
+						return
```

Comment:
It looks like what's happening here is that we're mounting a new child, then realizing we weren't supposed to do so and unmounting it. Is there a way we could avoid mounting it in the first place on line 117? Seems like we should move some of this logic up a bit.

Reply:
The mounting of the child node is what causes it to no longer be needed as we re-renter updateChildren for the same node in the tree again.